readthedocs/api/v3/permissions.py

```diff
@@ -1,6 +1,24 @@
 from rest_framework.permissions import BasePermission, IsAuthenticated
 
 from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.subscriptions.constants import TYPE_EMBED_API
+from readthedocs.subscriptions.models import PlanFeature
+
+
+class HasEmbedAPIAccess(BasePermission):
+
+    message = (
+        "Content embedding isn't available in your current plan. "
+        "Upgrade your subscription to enable this feature. "
+        "https://about.readthedocs.com/pricing/."
+    )
+
+    def has_permission(self, request, view):
+        project = view._get_project()
+        # TODO: what to do with external URLs?
+        if project and not PlanFeature.objects.has_feature(project, TYPE_EMBED_API):
```

Comment:
Not sure what to do in this case for the embedded API, should we just allow this? We have fixed number of allowed sites anyway, so..

https://github.com/readthedocs/readthedocs.org/blob/3fe7cc264f3d7079847b54dc5e95f50a798dbaf9/readthedocs/settings/base.py#L1036-L1041

Reply:
@stsewd can you ping someone else from the team about this particular change?

Reply:
Sorry, I think I posted about this in slack and not here.

We check features at the project or organization level, e.g "check if this organization or project have this feature", but the embed API v3 supports getting embedded content from external sites that aren't attached to a project or organization that we can check if they have the feature enabled or not.

But since that we restrict access to just a selected number of sites, I think we should be fine allowing those sites.

Reply:
Ah, I get it now, thanks :+1: 

I'm good with this, I just needed to understand what the permission handling was essentially supposed to do, but it helped to read your explanation and read the `message` :+1:

Reply:
Ah, I get it now, thanks :+1: 

I'm good with this, I just needed to understand what the permission handling was essentially supposed to do, but it helped to read your explanation and read the `message` :+1: